File: backend/app/schemas/financial.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class FinancialAnalysisCreate(BaseModel):
    simulation_id: uuid.UUID
    total_cost_fcfa: int
    annual_savings_fcfa: int
    senelec_tariff_applied: dict | None = None
    npv_fcfa: int | None = None
    irr_pct: Decimal | None = None
    payback_years: Decimal | None = None
    cashflow_25y: dict | None = None
    degradation_rate_pct: Decimal = Decimal("0.5")


class FinancialAnalysisRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    simulation_id: uuid.UUID
    total_cost_fcfa: int
    annual_savings_fcfa: int
    senelec_tariff_applied: dict | None = None
    npv_fcfa: int | None = None
    irr_pct: Decimal | None = None
    payback_years: Decimal | None = None
    cashflow_25y: dict | None = None
    degradation_rate_pct: Decimal
    created_at: datetime

File: frontend/src/types/financial.ts
export interface CashflowEntry {
  year: number;
  production_kwh: number;
  savings_fcfa: number;
  maintenance_fcfa: number;
  net_cashflow_fcfa: number;
  cumulative_fcfa: number;
}

export interface FinancialResult {
  id: string;
  simulation_id: string;
  total_cost_fcfa: number;
  annual_savings_year1_fcfa: number;
  payback_years: number | null;
  npv_fcfa: number;
  irr_pct: number | null;
  roi_pct: number;
  lcoe_fcfa_per_kwh: number;
  cashflow_25y: CashflowEntry[];
}

export interface FinancialRequest {
  total_cost_fcfa: number;
  annual_savings_fcfa: number;
  maintenance_annual_fcfa?: number;
  degradation_rate_pct?: number;
  discount_rate_pct?: number;
  inflation_rate_pct?: number;
}
